Mathlib/CategoryTheory/Triangulated/Opposite/Functor.lean

```diff
@@ -0,0 +1,195 @@
+      op_comp, Functor.op_map, Quiver.Hom.unop_op, unop_comp, Functor.map_comp, Iso.refl_hom,
+      Functor.map_id, comp_id, assoc, id_comp]
+    change _ = F.op.map ((opShiftFunctorEquivalence C 1).counitIso.inv.app (Opposite.op _)) ≫ _
+    rw [shift_map_op, map_shift_unop, map_opShiftFunctorEquivalence_counit_inv_app]
+    simp only [id_obj, op_obj, comp_obj, Quiver.Hom.unop_op, op_comp, op_unop, Quiver.Hom.op_unop,
+      assoc, Iso.inv_hom_id_app, comp_id, Iso.inv_hom_id_app_assoc,
+      opShiftFunctorEquivalence_inverse, opShiftFunctorEquivalence_functor]
+  · intro _ _ _
+    simp only [triangleOpEquivalence_functor, comp_obj, op_obj, mapTriangle_obj,
+      TriangleOpEquivalence.functor_obj, Triangle.mk_obj₃, Triangle.mk_obj₂, Triangle.mk_obj₁,
+      Triangle.mk_mor₂, Triangle.mk_mor₁, opShiftFunctorEquivalence_inverse,
+      opShiftFunctorEquivalence_functor, Triangle.mk_mor₃, op_comp, op_map, Quiver.Hom.unop_op,
+      unop_comp, comp_map, Triangle.isoMk_hom, Iso.refl_hom, triangleCategory_comp]
+    ext
+    · simp only [Triangle.mk_obj₁, TriangleMorphism.comp_hom₁,
+      TriangleOpEquivalence.functor_map_hom₁, Triangle.mk_obj₃, Quiver.Hom.unop_op,
+      mapTriangle_map_hom₃, Triangle.homMk_hom₁, comp_id, mapTriangle_map_hom₁, op_map, id_comp]
+    · simp only [Triangle.mk_obj₂, TriangleMorphism.comp_hom₂,
+      TriangleOpEquivalence.functor_map_hom₂, Quiver.Hom.unop_op, mapTriangle_map_hom₂,
+      Triangle.homMk_hom₂, comp_id, op_map, id_comp]
+    · simp only [Triangle.mk_obj₃, TriangleMorphism.comp_hom₃,
+      TriangleOpEquivalence.functor_map_hom₃, Quiver.Hom.unop_op, mapTriangle_map_hom₃,
+      Triangle.homMk_hom₃, comp_id, op_map, id_comp]
+      rfl
+
+/--
+If `F : C ⥤ D` commutes with shifts, this expresses the compatibility of `F.mapTriangle`
+with the equivalences `Pretriangulated.triangleOpEquivalence` on `C` and `D`.
+-/
+noncomputable def triangleOpEquivalence_inverse_naturality :
+    F.op.mapTriangle ⋙ (triangleOpEquivalence D).inverse ≅
+      (triangleOpEquivalence C).inverse ⋙ F.mapTriangle.op :=
+  (Functor.leftUnitor _).symm ≪≫ isoWhiskerRight (triangleOpEquivalence C).counitIso.symm _
+  ≪≫ Functor.associator _ _ _ ≪≫ isoWhiskerLeft (triangleOpEquivalence C).inverse
+  (Functor.associator _ _ _).symm ≪≫ isoWhiskerLeft _ (isoWhiskerRight
+  (triangleOpEquivalence_functor_naturality F).symm _) ≪≫ isoWhiskerLeft
+  (triangleOpEquivalence C).inverse (Functor.associator _ _ _) ≪≫ isoWhiskerLeft
+  (triangleOpEquivalence C).inverse (isoWhiskerLeft _ (triangleOpEquivalence D).unitIso.symm) ≪≫
+  isoWhiskerLeft _ (Functor.rightUnitor _)
+
+/-- If `F` is triangulated, so is `F.op`.
+-/
+lemma isTriangulated_op [F.IsTriangulated] : F.op.IsTriangulated where
+  map_distinguished T dT := by
+    rw [mem_distTriang_op_iff, ← Functor.comp_obj, ← distinguished_iff_of_iso
+      ((triangleOpEquivalence_inverse_naturality F).app T).unop]
+    exact F.map_distinguished _ ((mem_distTriang_op_iff _).mp dT)
```

Comment:
We may take advantage of definitional equalities here so as to avoid `← Functor.comp_obj`:
```suggestion
    rw [mem_distTriang_op_iff]
    exact Pretriangulated.isomorphic_distinguished _
      ((F.map_distinguished _ (unop_distinguished _ dT))) _ 
      (((opMapTriangleCompTriangleOpEquivalenceInverse F).symm.app T).unop)
```